packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -29,23 +29,29 @@ import 'platform_messages.dart';
 ///
 /// See: <https://flutter.dev/platform-channels/>
 class BasicMessageChannel<T> {
-  /// Creates a [BasicMessageChannel] with the specified [name] and [codec].
+  /// Creates a [BasicMessageChannel] with the specified [name], [codec] and [binaryMessenger].
   ///
-  /// Neither [name] nor [codec] may be null.
-  const BasicMessageChannel(this.name, this.codec);
+  /// None of [name], [codec], or [binaryMessenger] may be null.
+  const BasicMessageChannel(this.name, this.codec, [ this.binaryMessenger = defaultBinaryMessenger ])
```

Comment:
Can we make this a named parameter rather than an optional one?  e.g. `{ this.binaryMessenger = defaultBinaryMessenger }`?  This ends up getting ugly/confusing if we have to add more parameters later.

Reply:
Yes. I did it this way because `MethodChannel` has an optional argument for `this.codec` so I could only add `binaryMessenger` as another optional argument.